src/pageEditor/modListingPanel/modals/MoveCopyToModModal.tsx

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (C) 2024 PixieBrix, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useCallback, useMemo } from "react";
+import { Button, Modal } from "react-bootstrap";
+import SelectWidget from "@/components/form/widgets/SelectWidget";
+import { useDispatch, useSelector } from "react-redux";
+import { actions as editorActions } from "@/pageEditor/store/editor/editorSlice";
+import {
+  selectActiveModComponentFormState,
+  selectEditorModalVisibilities,
+  selectKeepLocalCopyOnCreateMod,
+  selectModMetadataMap,
+} from "@/pageEditor/store/editor/editorSelectors";
+import ConnectedFieldTemplate from "@/components/form/ConnectedFieldTemplate";
+import Form, {
+  type OnSubmit,
+  type RenderBody,
+  type RenderSubmit,
+} from "@/components/form/Form";
+import { object, string } from "yup";
+import { type RegistryId } from "@/types/registryTypes";
+import { assertNotNullish } from "@/utils/nullishUtils";
+
+type FormState = {
+  modId: RegistryId | null;
+};
+
+const initialFormState: FormState = {
+  modId: null,
+};
+
+const NEW_MOD_ID = "@new" as RegistryId;
+
+const formStateSchema = object({
+  modId: string().required(),
+});
+
+/**
+ * Modal to move/copy a mod component to an existing or new mod
+ */
+const MoveCopyToModModal: React.FC = () => {
```

Comment:
Very NIT, just because when I first read it I was confused: 
```suggestion
const MoveOrCopyToModModal: React.FC = () => {
```